e2e/static_count_test.go

```diff
@@ -1,129 +1,104 @@
+			{Flag: "proxy-server-port", Value: "8091"},
+			{Flag: "sync-interval", Value: "1s"},
+			{Flag: "sync-interval-cap", Value: "10s"},
+			{Flag: "sync-forever"},
+			{Flag: "probe-interval", Value: "1s"},
+			{Flag: "service-account-token-path", Value: "/var/run/secrets/tokens/konnectivity-agent-token"},
+			{Flag: "agent-identifiers", Value: "ipv4=${HOST_IP}"},
+			{Flag: "admin-bind-address", EmptyValue: true},
 		},
 	}
-	agentDeployment, _, err := renderTemplate("agent/deployment.yaml", agentDeploymentConfig)
+	agentDeployment, _, err = renderTemplate("agent/deployment.yaml", agentDeploymentConfig)
 	if err != nil {
-		t.Fatalf("could not render agent deployment: %v", err)
+		return nil, nil, fmt.Errorf("could not render agent deployment: %w", err)
+	}
+
+	return serverDeployment, agentDeployment, nil
+}
+
+func TestSingleServer_SingleAgent_StaticCount(t *testing.T) {
+	serverDeployment, agentDeployment, err := renderStaticCountDeployments(1, 1)
+	if err != nil {
+		t.Fatalf("could not render static count deployments: %v", err)
 	}
 
 	feature := features.New("konnectivity server and agent deployment with single replica for each")
-	feature.Setup(deployAndWaitForDeployment(serverDeployment))
-	feature.Setup(deployAndWaitForDeployment(agentDeployment))
-	feature.Assess("konnectivity server has a connected client", assertServersAreConnected(1, adminPort))
-	feature.Assess("konnectivity agent is connected to a server", assertAgentsAreConnected(1, adminPort))
+	feature = feature.Setup(createDeployment(agentDeployment))
+	feature = feature.Setup(createDeployment(serverDeployment))
+	feature = feature.Setup(waitForDeployment(agentDeployment))
+	feature = feature.Setup(waitForDeployment(serverDeployment))
+	feature = feature.Assess("konnectivity server has a connected client", assertServersAreConnected(1))
+	feature = feature.Assess("konnectivity agent is connected to a server", assertAgentsAreConnected(1))
+	feature = feature.Assess("agents correctly count 1 server", assertAgentKnownServerCount(1))
+	feature = feature.Teardown(deleteDeployment(agentDeployment))
+	feature = feature.Teardown(deleteDeployment(serverDeployment))
+
+	testenv.Test(t, feature.Feature())
 }
 
 func TestMultiServer_MultiAgent_StaticCount(t *testing.T) {
-	adminPort := 8093
 	replicas := 3
-
-	serverDeploymentConfig := DeploymentConfig{
-		Replicas: replicas,
-		Image:    *serverImage,
-		Args: []KeyValue{
-			{"log-file", "/var/log/konnectivity-server.log"},
-			{"logtostderr", "true"},
-			{"log-file-max-size", "0"},
-			{"uds-name", "/etc/kubernetes/konnectivity-server/konnectivity-server.socket"},
-			{Key: "delete-existing-uds-file"},
-			{"cluster-cert", "/etc/kubernetes/pki/apiserver.crt"},
-			{"cluster-key", "/etc/kubernetes/pki/apiserver.key"},
-			{"server-port", "8090"},
-			{"agent-port", "8091"},
-			{"health-port", "8092"},
-			{"admin-port", strconv.Itoa(adminPort)},
-			{"keepalive-time", "1h"},
-			{"mode", *connectionMode},
```

Comment:
e2e/main_test.go:38:2: var `connectionMode` is unused (unused)
	connectionMode = flag.String("mode", "grpc", "Connection mode to use during e2e tests.")